### PR TITLE
tigmint: bump revision for python 3.7

### PR DESCRIPTION
**Pull request by suujia**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----
 fixes `Missing libraries: libpython3.6m.so.1.0`

### CODE REVIEW

**Comment by suujia** [comment]
Oh, seems to be the same pysam error found in: https://github.com/brewsci/homebrew-bio/pull/351
```
  Failed building wheel for pybedtools
  Running setup.py clean for pybedtools
  Running setup.py bdist_wheel for pysam: started
  Running setup.py bdist_wheel for pysam: finished with status 'error'
```

**Comment by sjackman** [comment]
Could you please squash this PR to a single commit? In general, you can use `git commit --amend` to keep the PR to a single commit, unless you intentionally want to split the PR into multiple commits, for example when a PR modifies multiple formulae.

**Comment by suujia** [comment]
linux: 
```
pysam/libchtslib.c:15313:12: error: too many arguments to function ‘(PyObject * (*)(PyObject *, PyObject * const*, Py_ssize_t))meth’
         return (*((__Pyx_PyCFunctionFast)meth)) (self, args, nargs, NULL);
                ^
    error: command 'gcc-5' failed with exit status 1
```
macos:
```
  Running setup.py bdist_wheel for pybedtools: started
  Running setup.py bdist_wheel for pybedtools: finished with status 'error'
  Complete output from command /usr/local/opt/python/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/private/tmp/pip-install-znu2j9ov/pybedtools/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /private/tmp/pip-wheel-6wsaqcxb --python-tag cp37:
```
Hmm, pybedtools and pysam are both installed from the requirements.txt files but I tested it by adding resource downloads for both of them in this formula. Still not working with the same messages

**Comment by sjackman** [comment]
This poster reports that `pip3 install pysam` fails but
```sh
pip3 install https://github.com/pysam-developers/pysam/archive/master.zip
```
does in fact work.
See https://github.com/pysam-developers/pysam/issues/697#issuecomment-402735807
Instead of using the `requirements.txt` provided by `tigmint`, you could try…
```sh
pip3 install intervaltree pybedtools statistics
pip3 install https://github.com/pysam-developers/pysam/archive/master.zip
```

**Comment by suujia** [comment]
I added a resource for cython in as well but still receive this error message: 
```
ValueError: no cython installed, but can not find pysam/libchtslib.c.Make sure that cython is installed when building from the repository
----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/tmp/pip-req-build-jd1a3_qo/
```

**Comment by suujia** [comment]
hmm I've tried plain pip3 install cython and now trying to install by resource. Both output this message: 
```
ValueError: no cython installed, but can not find pysam/libchtslib.c.Make sure that cython is installed when building from the repository
```

File: Formula/tigmint.rb [hunk 1]
```diff
@@ -17,8 +17,25 @@ class Tigmint < Formula
...
   def install
-    system "pip3", "install", "--prefix=#{libexec}", "-r", "requirements.txt"
+    %w[cython pysam].each do |r|
+      resource(r).stage do
+        system "python3", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
+
+    tools = %w[intervaltree pybedtools statistics]
+    system "pip3", "install", *tools
```
Comment (sjackman) [inline Formula/tigmint.rb]: Try…
```ruby
system "pip3", "install", "--prefix=#{libexec}", "intervaltree", "pybedtools", "statistics"
```

File: Formula/tigmint.rb [hunk 2]
```diff
@@ -17,8 +17,25 @@ class Tigmint < Formula
...
+    url "https://files.pythonhosted.org/packages/21/89/ca320e5b45d381ae0df74c4b5694f1471c1b2453c5eb4bac3449f5970481/Cython-0.28.5.tar.gz"
+    sha256 "b64575241f64f6ec005a4d4137339fb0ba5e156e826db2fdb5f458060d9979e0"
+  end
+
+  resource "pysam" do
+    url "https://github.com/pysam-developers/pysam/archive/v0.15.0.1.tar.gz"
+    sha256 "b169ffbe0efb39fd193779e5982da1de86e392dfe66c6bc49d79aa34fe58b46b"
+  end
+
   def install
```
Comment (sjackman) [inline Formula/tigmint.rb]: Try adding…
```
    xy = Language::Python.major_minor_version "python3"
    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
```

File: Formula/tigmint.rb [hunk 3]
```diff
@@ -17,8 +17,25 @@ class Tigmint < Formula
...
-    system "pip3", "install", "--prefix=#{libexec}", "-r", "requirements.txt"
+    %w[cython pysam].each do |r|
+      resource(r).stage do
+        system "python3", *Language::Python.setup_install_args(libexec/"vendor")
```
Comment (sjackman) [inline Formula/tigmint.rb]: Try removing `/"vendor"`

**Comment by suujia** [comment]
Hi @sjackman, I believe this is the last tool that was broken from the last batch. Do you have any suggestions for what I can do? It seems its still a `pybedtools` tool error, and how it's having troubles with cython. I tried pip3 install and build by resource 
I've tried pysam by source and by pip3 install master branch 

```
    # pysam: no cython available - using pre-compiled C
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-v0fbfdd6/setup.py", line 175, in <module>
        .format(fn))
    ValueError: no cython installed, but can not find pysam/libchtslib.c.Make sure that cython is installed when building from the repository
```

Doesn't seem to be able to notice cython. Similar problem to: https://github.com/brewsci/homebrew-bio/pull/351
Except circlator uses virtualenv_install_with_resources https://github.com/Linuxbrew/brew/blob/c8d39cf0b8505b7038398d7c4010d080b226cb54/Library/Homebrew/language/python.rb#L138

**Comment by sjackman** [comment]
This was a tricky one!
```
pip3 install --prefix=/home/linuxbrew/.linuxbrew/Cellar/tigmint/1.1.2_1/libexec cython
Collecting cython
  Downloading https://files.pythonhosted.org/packages/45/13/f3b479013bd3ca96696702fa823ff3d8f81a445dde9a5b2dfee361404bfe/Cython-0.28.5-cp37-cp37m-manylinux1_x86_64.whl (3.4MB)
Installing collected packages: cython
  The scripts cygdb, cython and cythonize are installed in '/home/linuxbrew/.linuxbrew/Cellar/tigmint/1.1.2_1/libexec/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed cython-0.28.5
```
I found I needed to modify both `PATH` and `PYTHONPATH` and install `pysam` and `pybedtools` from their GitHub repo.

```ruby
  def install
    ENV.prepend_path "PATH", libexec/"bin"
    xy = Language::Python.major_minor_version "python3"
    ENV.prepend_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
    system "pip3", "install", "--prefix=#{libexec}", 
      "cython",
      "git+https://github.com/pysam-developers/pysam.git",
      "git+https://github.com/daler/pybedtools.git"
    system "pip3", "install", "--prefix=#{libexec}", "-r", "requirements.txt"
    bin.install Dir["bin/*"]
    bin.env_script_all_files libexec/"bin", :PYTHONPATH => Dir[libexec/"lib/python*/site-packages"].first
  end
```

**Comment by suujia** [comment]
Perfect, Thank you @sjackman !! 
CI actually returned error for pip3 install cython so I just added the resource and it was good 
https://circleci.com/gh/brewsci/homebrew-bio/3825

**Comment by sjackman** [comment]
Merged. Thank you, Susie! 🏆

**Pull request closed without merging**